test/multicall/Multicall.js

```diff
@@ -0,0 +1,328 @@
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(n * INPUT_ELEMENT_LENGTH);    // data is 2.6 kb
+
+            const res = await multicallProcessLongInput(callData);
+            expect(res).to.exist;
+            expect(res.length).to.eq(n);
+            for (let i = 0; i < n; i++) {
+                expect(res[i].success).to.eq(true)
+                expect(res[i].returnData).to.eq(RESULT_FIVE)
+            }
+        });
+
+        it('should be able to aggregate 1000 calls to processLongInput', async function () {
+            const n = 1000;
+            const {callData, data} = prepareLongInputData(n, LONG_INPUT_ABI, LONG_INPUT_PARAMS);
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(n * INPUT_ELEMENT_LENGTH);    // data is 260 kb
+
+            const res = await multicallProcessLongInput(callData);
+            expect(res).to.exist;
+            expect(res.length).to.eq(n);
+            for (let i = 0; i < n; i++) {
+                expect(res[i].success).to.eq(true)
+                expect(res[i].returnData).to.eq(RESULT_FIVE)
+            }
+        });
+
+        it('should NOT be able to aggregate 5000 calls to processLongInput', async function () {
+            const n = 5000;
+            const {callData, data} = prepareLongInputData(n, LONG_INPUT_ABI, LONG_INPUT_PARAMS);
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(1330000);    // data is 1,3 mb
+
+            // Input size is larger than 1 mb and the call is rejected by the relay
+            let hasError = false;
+            try {
+                const res = await multicallProcessLongInput(callData);
+            } catch (e) {
+                hasError = true;
+            }
+
+            expect(hasError).to.eq(true);
+        });
+
+
+        it('should be able to aggregate 11 calls to processLongInput and handles a revert', async function () {
+            let hasError = false;
+            try {
+                const {callData, data} = prepareLongInputData(10, LONG_INPUT_ABI, LONG_INPUT_PARAMS, true);
+                const res = await multicallProcessLongInput(callData);
+            } catch (e) {
+                hasError = true;
+                expect(e.data).to.exist;
+                expect(e.data).to.eq(REVERT_REASON_DATA);
+            }
+
+            expect(hasError).to.eq(true);
+        });
+    })
+
+    describe('static calls with large output', async function () {
+        it('should be able to aggregate 10 calls to processLongOutput and handle 13 kb of output data', async function () {
+            const n = 10;
+            const res = await multicallProcessLongOutput(n);
+            expect(res).to.exist;
+            expect(res.length).to.eq(n);
+            const bytes = getOutputLengthInBytes(res.map(r => r.returnData));
+            expect(bytes).to.gte(13000); // 13 kb
```

Comment:
I still don't get where the 13000 or 820000 comes from - seems to be an interaction between what's going on in the contract and how multicallProcessLongOutput is building the multicall using incrementing values of `n` for each call?  Lot's of moving parts to have to know all at once ...

Reply:
The numbers here are arbitrary and not that important. All of those tests are intended to test the input and output size limits of pure and non-pure methods using a multicall contract. Most of those limits are actually not intentional and they may be different of every network because of the differences in gas price.